### ISSUE TITLE
`uninlined_format_args` is documented as part of the `style` lint group, but it is not

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust-clippy#10082]
### Summary

[The changelog linked from Rust 1.66 release notes](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-166) says that the lint `uninlined_format_args` was added in Rust 1.66. Following the link from there to [the lint list](https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args) says that the lint is in the `style` group and warn by default. However, this does not seem to be the case _in the stable release_; it does not warn unless set explicitly.

### Lint Name

uninlined_format_args

### Reproducer

I tried this code:

```rust
pub fn foo(x: &str) -> String {
    format!("Foo {}", x)
}
```

I expected to see this happen: A `uninlined_format_args` warning.

Instead, this happened: No warnings, unless I add `#[warn(clippy::uninlined_format_args)]`.

Adding `#[warn(clippy::style)]` also has no effect even though the lint claims to be part of the `style` group.

### Version

```text
rustc 1.66.0 (69f9c33d7 2022-12-12)
binary: rustc
commit-hash: 69f9c33d71c871fc16ac445211281c6e7a340943
commit-date: 2022-12-12
host: x86_64-apple-darwin
release: 1.66.0
LLVM version: 15.0.2
```

Note: This does not reproduce in nightly, including via Rust Playground —— the lint triggers as expected — suggesting that the problem is that the change to make this lint part of the `style` group missed being included in the stable release even though the changelog claims it did.

**Comment by Rudxain** [comment]
Same issue here. rust-analyzer reports it as part of `pedantic` group instead. Which caused it to trigger on my code, despite not having `style` enabled

**Comment by Alexendoo** [comment]
It is in pedantic for 1.66, the documentation is for the master branch where it has been moved to style. Ideally we would link to the 1.66 docs in the changelog, but they don't exist yet

**Comment by Rudxain** [comment]
Why is this still open? The docs are correct now
